src/App.tsx

```diff
@@ -17,18 +73,43 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                onChangeQuery={setQuery}
+                selectedStatus={selectedStatus}
+                onSelectStatus={setSelectedStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length && (
+                <TodoList
+                  todos={visibleTodos}
+                  onSelectTodo={handleSelectTodo}
+                  selectedId={selectedTodo?.id}
+                />
+              )}
+              {!todos.length && <Loader />}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {hasError
+        && (
+          <Error
+            hasError={hasError}
+            onModalClose={() => setSelectedTodo(undefined)}
```

Comment:
Undefined mostly used when there is no value at all. But when we want to reset or clear something, we use null. In practice it'll work the same, but that's how most developers write, so let's keep the style.